lib/flatpak-installation.c

```diff
@@ -1168,23 +1178,28 @@ progress_cb (OstreeAsyncProgress *progress, gpointer user_data)
                                   formatted_bytes_sec, formatted_bytes_transferred,
                                   formatted_total);
 
-          new_progress = (100 * bytes_transferred) / total_delta_part_size;
+
+          if (outstanding_extra_data)
+            new_progress = 10 + (35 * bytes_transferred) / total_delta_part_size;
+          else
+            new_progress = (100 * bytes_transferred) / total_delta_part_size;
         }
       else if (outstanding_metadata_fetches)
         {
           /* At this point we don't really know how much data there is, so we have to make a guess.
            * Since its really hard to figure out early how much data there is we report 1% until
            * all objects are scanned. */
 
-          new_progress = 1;
           estimating = TRUE;
 
           g_string_append_printf (buf, "Receiving metadata objects: %u/(estimating) %s/s %s",
                                   metadata_fetched, formatted_bytes_sec, formatted_bytes_transferred);
+
+          /* Go up to 10% until the metadata is all fetched */
+          new_progress = (10 * metadata_fetched) / (outstanding_metadata_fetches + metadata_fetched);
```

Comment:
This can go backwards. For instance, initially metadata_fetched is say 1, and outstanding fetches is 1, so, progress is  5%, then when we get the 1 extra fetch. However, it will converge on the right result eventually, so maybe that is ok.

However, i think we really want some protection against divide by zero here.